packages/app/components/Settings/ToggleFeaturesPage.tsx

```diff
@@ -0,0 +1,189 @@
+import { QuestionCircleOutlined } from '@ant-design/icons'
+import { API } from '@koh/api-client'
+import { Form, Spin, message, Switch, Tooltip } from 'antd'
+import { ReactElement } from 'react'
+import styled from 'styled-components'
+import useSWR, { mutate } from 'swr'
+
+type ToggleFeaturesPageProps = { courseId: number }
+
+const ToggleFeaturesPageComponent = styled.div`
+  width: 90%;
+  margin-left: auto;
+  margin-right: auto;
+  padding-top: 50px;
+`
+
+const CustomFormItem = styled(Form.Item)`
+  padding-bottom: 1rem;
+  margin-bottom: 1rem;
+
+  font-size: 1rem;
+  line-height: 0.5rem;
+
+  &:last-child {
+    padding-bottom: 0;
+    margin-bottom: 0;
+  }
+`
+
+const CustomSwitch = styled(Switch)`
+  margin-right: 1rem;
+`
+
+export default function ToggleFeaturesPage({
+  courseId,
+}: ToggleFeaturesPageProps): ReactElement {
+  const { data: courseFeatures } = useSWR(
+    `${courseId}/features`,
+    async () => await API.course.getCourseFeatures(courseId),
+  )
+  if (!courseFeatures) {
+    return <Spin tip="Loading..." size="large" />
+  } else {
+    return (
+      <ToggleFeaturesPageComponent>
+        <h2>Enable/Disable Features for this Course</h2>
+
+        <Form className="ml-2">
+          <CustomFormItem>
+            <CustomSwitch
+              defaultChecked={courseFeatures.asyncQueueEnabled}
+              onChange={async (e) => {
+                await API.course
```

Comment:
This seems like a repeated code, consider moving it to the function and checking the selected features through the switch statement?

Reply:
moved it to a function, thanks